maestro-ios-xctest-runner/maestro-driver-iosUITests/ViewHierarchyHandleTests.swift

```diff
@@ -0,0 +1,70 @@
+import XCTest
+import FlyingFox
+
+final class ViewHierarchyHandlerTests: XCTestCase {
+    
+    override func setUpWithError() throws {
+        continueAfterFailure = false
+        Task {
+            try await startFlyingFoxServer()
+        }
+    }
+    
+    func startFlyingFoxServer() async throws {
+        let server = HTTPServer(port: 8080)
+        await server.appendRoute("hierarchy", to: ViewHierarchyHandler())
+        try await server.run()
+    }
+    
+    func testAppOffsetAdjustsCorrectly() async throws {
+        // given
+        guard let url = URL(string: "http://localhost:8080/hierarchy") else {
+            throw NSError(domain: "XCTestError", code: 1, userInfo: [NSLocalizedDescriptionKey: "Failed to construct URL"])
+        }
+        var request = URLRequest(url: url)
+        request.httpMethod = "POST"
+        request.httpBody = try JSONEncoder().encode(ViewHierarchyRequest(appIds: [], excludeKeyboardElements: false))
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+        
+        let springboardApp = await XCUIApplication(bundleIdentifier: "com.apple.springboard")
+        let testApp = await XCUIApplication(bundleIdentifier: "org.wikimedia.wikipedia")
+        let springboardFrame = await springboardApp.frame
+        let testAppFrame = await testApp.frame
+        
+        let offsetX = springboardFrame.width - testAppFrame.width
```

Comment:
it would make more sense to have the expected numbers hardcoded in the test - the way it is right now, this could mean that the actual implementation is wrong, and the test would quietly accept it?

Reply:
In general, hardcoding expected values is a solid way to catch regressions. But in this case, we avoided that because the offset between the springboard and app frames varies by simulator/device (e.g., iPhone 14 Pro vs SE). Hardcoding values like +30 or +12 would require forcing a specific destination, which we want to avoid for test portability.

We're asserting that the actual app frame reflects the runtime-calculated offset between springboard and app dimensions. So if the implementation is wrong (e.g. offset isn’t applied), this test still fails. The test is dynamic but still strict since we assert the calculation here.